pkg/spoke/spokeagent.go

```diff
@@ -399,23 +440,39 @@ func (o *SpokeAgentOptions) getOrGenerateClusterAgentNames() (string, string) {
 		// and then load the cluster name from the mounted secret
 		clusterNameFilePath := path.Join(o.HubKubeconfigDir, hubclientcert.ClusterNameFile)
 		clusterNameBytes, err := ioutil.ReadFile(path.Clean(clusterNameFilePath))
-		if err != nil {
-			// generate random cluster name if faild
-			clusterName = generateClusterName()
-		} else {
+		switch {
+		case len(clusterNameInCert) > 0:
+			// use cluster name loaded from the tls certification
+			clusterName = clusterNameInCert
+			if err == nil && clusterNameInCert != string(clusterNameBytes) {
+				klog.Warningf("Cluster name in certification %q is different with the one in the mounted secret %q", clusterNameInCert, string(clusterNameBytes))
+			}
+		case err == nil:
+			// use cluster name load from the mounted secret
 			clusterName = string(clusterNameBytes)
+		default:
+			// generate random cluster name
+			clusterName = generateClusterName()
 		}
 	}
 
 	// try to load agent name from the mounted secret
 	agentNameFilePath := path.Join(o.HubKubeconfigDir, hubclientcert.AgentNameFile)
 	agentNameBytes, err := ioutil.ReadFile(path.Clean(agentNameFilePath))
 	var agentName string
-	if err != nil {
-		// generate random agent name if faild
-		agentName = generateAgentName()
-	} else {
+	switch {
+	case len(agentNameInCert) > 0:
+		// use agent name loaded from the tls certification
+		agentName = agentNameInCert
+		if err == nil && agentNameInCert != string(agentNameBytes) {
```

Comment:
i am not sure we need err==nil here, I think we output the warning whenever it is different

Reply:
It makes sense. Whenever the cluster/agent name in cert is used, we should output a warning message.